### PR TITLE
Support Azure Blob Storage key prefixes

### PR DESCRIPTION
**Pull request by FooBarWidget**
Similar to how we support key prefixes in S3 and Google Cloud Storage.

Introduces `SCCACHE_AZURE_KEY_PREFIX`.

### CODE REVIEW

File: src/cache/azure.rs [hunk 1]
```diff
@@ -25,16 +25,20 @@ use crate::errors::*;
...
 
 impl AzureBlobCache {
-    pub fn new() -> Result<AzureBlobCache> {
-        let credentials = match EnvironmentProvider.provide_credentials() {
-            Ok(creds) => creds,
-            Err(err) => bail!(
-                "Could not find Azure credentials in the environment: {}",
-                err
-            ),
+    pub fn new(credentials: Option<AzureCredentials>, key_prefix: &str) -> Result<AzureBlobCache> {
```
Comment (mitchhentges) [inline src/cache/azure.rs:32]: Good call on making `credentials` be provided by the consumer to make your tests possible to write.

It's a little funky seeing the `None` get provided into `AzureBlobCache` in `storage_from_config()` down below: perhaps it's worth trying one of the following two options:
1. Make `credentials` mandatory, and move the default `EnvironmentProvider.provide_credentials()` up to the caller
2. (My preferred option): Make two separate functions here: `pub fn new(credentials: AzureCredentials, key_prefix: &str)` and `pub fn new_with_environment_credentials(key_prefix: &str)`.

Let me know what you think :)

File: src/cache/azure.rs [hunk 2]
```diff
@@ -70,18 +84,27 @@ impl Storage for AzureBlobCache {
...
             .await
             .context("Failed to put cache entry in Azure")?;
 
         Ok(start.elapsed())
     }
 
     fn location(&self) -> String {
-        format!("Azure, container: {}", self.container)
+        format!(
+            "Azure, container: {}, {}",
```
Comment (mitchhentges) [inline src/cache/azure.rs:100]: Let's match the format used by GCS.
```suggestion
            "Azure, container: {}, key_prefix: {}",
```

**Review by mitchhentges — APPROVED** [review]
Nice work, and thanks for the contribution!
I'll let this sit for a day or two (or until your response + potential update), then land afterwards.
Cheers :beers:

File: src/cache/azure.rs (hunk 1 above)
Comment (FooBarWidget) [reply src/cache/azure.rs:32]: I'm fine with either, so I'll implement the second.

Comment (FooBarWidget) [reply src/cache/azure.rs:32]: ~I'm fine with either, so I'll implement the second.~
On second thought, I think the first one is better going forward. The problem with the second one is that, as more options get introduced for AzureStorage, you need to duplicate all parameters twice.

**Review by mitchhentges — APPROVED** [review]
Great, thanks again!

**Comment by mitchhentges** [comment]
Working towards it, see #1019!

**Pull request merged**